### PR TITLE
更新ModBuilder，修复HookManager，删除部分过时代码

### PR DESCRIPTION
**Pull request by Solaestas**

### CODE REVIEW

**Comment by Setnour6** [comment]
I used the blue ribbon bayonet right-click attack and it crashed the game, even if the debug mode config in-game is turned off. I am not sure if that is what I am supposed to be testing.

**Review by Omnielement — APPROVED** [review]

File: Sources/Everglow.Function/VFX/VFXManager.cs [hunk 1]
```diff
@@ -10,16 +10,16 @@ namespace Everglow.Commons.VFX;
...
+	[
 		CodeLayer.PreDrawFilter,
 		CodeLayer.PostDrawProjectiles,
 		CodeLayer.PostDrawTiles,
 		CodeLayer.PostDrawDusts,
 		CodeLayer.PostDrawBG,
 		CodeLayer.PostDrawPlayers,
 		CodeLayer.PostDrawNPCs,
-	};
+	];
```
Comment (Setnour6) [inline Sources/Everglow.Function/VFX/VFXManager.cs:22]: Error CS1525: Invalid expression term '['	in Everglow.Function	Line 13
Error CS0443: Syntax error; value expected	in Everglow.Function	Line 21

File: Sources/Everglow.Function/VFX/VFXManager.cs [hunk 2]
```diff
@@ -38,7 +38,7 @@ public VFXManager(IHookManager hookManager, IMainThreadContext mainThread)
 	{
 		foreach (var layer in drawLayers)
 		{
-			visuals[layer] = new List<IVisualCollection>();
+			visuals[layer] = [];
```
Comment (Setnour6) [inline Sources/Everglow.Function/VFX/VFXManager.cs:41]: Error CS1525: Invalid expression term '['	in Everglow.Function	Line 41
Error CS0443: Syntax error; value expected	in Everglow.Function	Line 41

**Review by Setnour6 — CHANGES_REQUESTED** [review]

**Comment by Setnour6** [comment]
I can take back the "request changes" and allow this PR to be merged if you really want this PR to be merged now. I just cannot build and test the mod because of the errors

**Review by Setnour6 — CHANGES_REQUESTED** [review]
On further testing, anything that uses bloom or special VFX (Bayonets attacks, New Magic Books, Tooth Spear) does not work or fails to work properly.

File: Sources/Everglow.Function/VFX/VFXManager.cs (hunk 1 above)
Comment (Solaestas) [reply Sources/Everglow.Function/VFX/VFXManager.cs]: Unexpectedly, it needs dotnet7(or higher?) to support C#12
I already change the Lang Version into C#11

**Review by Setnour6 — APPROVED** [review]
Everything works now

**Comment by Omnielement** [comment]
> Everything works now

you can also try to update your dotnot.The problem is your out of dated c# can not compile the code with new grammar.

**Comment by Setnour6** [comment]
> > Everything works now
> 
> you can also try to update your dotnot.The problem is your out of dated c# can not compile the code with new grammar.

I have .NET SDK 7.0.203 (x64) installed but I forgot how to use it

**Pull request merged**